frontend/public/components/cluster-settings/cluster-settings.tsx

```diff
@@ -247,6 +255,144 @@ const ChannelHeader: React.FC<{}> = () => {
   );
 };
 
+const Channel: React.FC<ChannelProps> = ({ children, endOfLife, upToDate }) => {
+  return (
+    <div
+      className={classNames('co-channel', {
+        'co-channel--up-to-date': upToDate,
+        'co-channel--end-of-life': endOfLife,
+      })}
+    >
+      {children}
+    </div>
+  );
+};
+
+const ChannelLine: React.FC<ChannelLineProps> = ({ children, start }) => {
+  return (
+    <li className={classNames('co-channel-line', { 'co-channel-start': start })}>{children}</li>
+  );
+};
+
+const ChannelName: React.FC<ChannelNameProps> = ({ children, current }) => {
+  return (
+    <span
+      className={classNames('co-channel-name', {
+        'co-channel-name--current': current,
+      })}
+    >
+      {children}
+    </span>
+  );
+};
+
+const ChannelPath: React.FC<ChannelPathProps> = ({ children, current }) => {
+  return (
+    <ul
+      className={classNames('co-channel-path', {
+        'co-channel-path--current': current,
+      })}
+    >
+      {children}
+    </ul>
+  );
+};
+
+const ChannelVersion: React.FC<ChannelVersionProps> = ({ children, current }) => {
+  return (
+    <span
+      className={classNames('co-channel-version', {
+        'co-channel-version--current': current,
+      })}
+    >
+      {children}
+    </span>
+  );
+};
+
+const ChannelVersionDot: React.FC<ChannelVersionDotProps> = ({ current }) => {
+  return (
+    <div
+      className={classNames('co-channel-version-dot', {
+        'co-channel-version-dot--current': current,
+      })}
+    ></div>
+  );
+};
+
+const getChannelPrefix = (channel: string): string => channel.split('-')[0];
```

Comment:
We probably want to include the `-` in the prefix if we're doing a substring match. It won't be a problem with the channels we have today, but in theory if we ever have channels like `fast-` and `faster-`, we'd have some false positives unless we include the `-` check below.

We'd have problems with a value like `release-candidate-4.6`, too. (We don't have a channel with this name, but it'd be good to handle any value in case we ever add new ones.)

I'd suggest a regex, which is a little safer.

```ts
const splitChannel = (channel: string) => {
  const parsed = /^(.*)-(\d\.\d+)$/.exec(channel);
  return parsed
    ? { prefix: parsed[1], version: parsed[2] }
    : null;
}
```

This uses capture groups to get the prefix and version and returns null for anything that doesn't match the expected format. Then you can do something like this to filter.

```ts
const currentPrefix = splitChannel(currentChannel)?.prefix;
const similarChannels = channels.filter((channel: string) => {
  return currentPrefix && splitChannel(channel)?.prefix === currentPrefix;
});
```